flatten-docs.py

```
@@ -0,0 +1,287 @@
+    resolved_target = (current_dir / url).resolve()
+
+    # Make path relative to version_dir or DOCS_ROOT
+    try:
+        relative_target = resolved_target.relative_to(version_dir)
+        base_dir = version_dir
+    except ValueError:
+        try:
+            relative_target = resolved_target.relative_to(DOCS_ROOT_RESOLVED)
+            base_dir = DOCS_ROOT_RESOLVED
+        except ValueError:
+            raise ValueError(f"Link target {resolved_target} is outside docs tree")
```

Ruff (0.14.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)